controllers/openstackmachine_controller.go

```diff
@@ -492,9 +492,9 @@ func machineToInstanceSpec(openStackCluster *infrav1.OpenStackCluster, machine *
 	instanceSpec.SecurityGroups = openStackMachine.Spec.SecurityGroups
 	if openStackCluster.Spec.ManagedSecurityGroups {
 		var managedSecurityGroup string
-		if util.IsControlPlaneMachine(machine) {
+		if util.IsControlPlaneMachine(machine) && openStackCluster.Status.ControlPlaneSecurityGroup != nil {
```

Comment:
femtonit: I'd probably have gone with nested if statements, just to clarify that this is the block where all controller-related $things happen. As it stands, this is now the block where all controller-related $things happen *if* `ControlPlaneSecurityGroup` is set. The below conditional ensures we won't go into that block if if this is *not* set but it takes a sec